Sources/Navigate/Navigate.swift

```diff
@@ -1,2 +1,9 @@
-// The Swift Programming Language
-// https://docs.swift.org/swift-book
+import SwiftUI
+
+/// A type that represents a possible NavigationDestination in your app
+public protocol NavigationDestination<Body>: Hashable, Identifiable {
```

Comment:
I think this should be `@MainActor` to be safe for Swift 6 mode and also because it is a UI layer thing